docs/collection.md

```diff
@@ -311,3 +329,308 @@ $unsorted = new ArrayList(['x', 'c', 'a', 't', 'm']);
 $sorted = $unsorted->sort(new ComparableComparator());
 print_r($sorted); // ['a', 'c', 'm', 't', 'x']
 ```
+
+### Remove
+
+You can remove one or more elements from a collection, via the [remove()](api/phootwork/collection/ArrayList.html#method_remove)
+method:
+
+```php
+<?php declare(strict_types=1);
+use phootwork\collection\ArrayList;
+
+$fruits = new ArrayList(['apple', 'banana', 'pine', 'banana', 'ananas']);
+
+$fruits->remove('banana');
+$fruits->remove('banana', 'ananas', 'pine');
+
+$fruits->toArray(); // ['apple']
+```
+
+### More Methods
+
+You can find more *ArrayList* methods into the [api documentation](api/phootwork/collection/ArrayList.html).
+
+## Set
+
+The [Set](api/phootwork/collection/Set.html) collection is composed by *unique* elements. It shares much methods with `ArrayList` except those which manipulate indexes.
+When you use a `Set` collection you really don't care about indexes.
+
+As usual, you can create a Set collection by passing an array or an [Iterator](https://www.php.net/manual/en/class.iterator.php)
+object to the constructor:
+
+```php
+<?php declare(strict_types=1);
+use phootwork\collection\Set;
+
+$fruits = new Set(['apple', 'pear', 'peach']);
+```
+
+Since the collection is composed by unique elements, any duplicates are ignored: 
+
+```php
+<?php declare(strict_types=1);
+use phootwork\collection\Set;
+
+$fruits = new Set(['apple', 'pear', 'peach', 'pear', 'banana', 'apple']);
+
+$fruits->toArray(); // ['apple', 'pear', 'peach', 'banana']
+```
+
+### Addition
+
+You can add one or more elements to the *Set* via the [add()](api/phootwork/collection/Set.html#method_add) method:
+
+```php
+<?php declare(strict_types=1);
+use phootwork\collection\Set;
+
+$fruits = new Set(['apple', 'pear', 'peach']);
+
+$fruits->add('banana');
+$fruits->add('pineapple', ['kiwi', 'lychee'], 'dorian');
+
+$fruits->toArray(); // ['apple', 'pear', 'peach', 'pineapple', ['kiwi', 'lychee'], 'dorian']
+
+// Since the Set elements are unique, the instruction below doesn't modify the collection
+$fruits->add('apple', 'peach');
+
+$fruits->toArray(); // ['apple', 'pear', 'peach', 'pineapple', ['kiwi', 'lychee'], 'dorian']
+```
+
+!!! note
+    Set doesn't have the [insert()](#addition) method.
```

Comment:
The link target will not work as expected. With markdown, each headline should only exist _once_ per document (identified by the title itself). If there are more, the link targets are not uniquly clear - keep in mind.

Reply:
Good catch! I wrong the link.